okhttp/src/main/java/okhttp3/internal/http/HttpDate.kt

```diff
@@ -65,55 +63,51 @@
       "EEE, dd-MM-yyyy HH:mm:ss z",
 
       /* RI bug 6641315 claims a cookie of this format was once served by www.yahoo.com */
-      "EEE MMM d yyyy HH:mm:ss z",
-  };
+      "EEE MMM d yyyy HH:mm:ss z")
 
-  private static final DateFormat[] BROWSER_COMPATIBLE_DATE_FORMATS =
-      new DateFormat[BROWSER_COMPATIBLE_DATE_FORMAT_STRINGS.length];
+  private val BROWSER_COMPATIBLE_DATE_FORMATS = arrayOfNulls<DateFormat>(BROWSER_COMPATIBLE_DATE_FORMAT_STRINGS.size)
 
-  /** Returns the date for {@code value}. Returns null if the value couldn't be parsed. */
-  public static Date parse(String value) {
-    if (value.length() == 0) {
-      return null;
+  /** Returns the date for [value]. Returns null if the value couldn't be parsed.  */
+  @JvmStatic
+  fun parse(value: String): Date? {
+    if (value.isEmpty()) {
+      return null
     }
 
-    ParsePosition position = new ParsePosition(0);
-    Date result = STANDARD_DATE_FORMAT.get().parse(value, position);
-    if (position.getIndex() == value.length()) {
+    val position = ParsePosition(0)
+    var result = STANDARD_DATE_FORMAT.get().parse(value, position)
+    if (position.index == value.length) {
       // STANDARD_DATE_FORMAT must match exactly; all text must be consumed, e.g. no ignored
       // non-standard trailing "+01:00". Those cases are covered below.
-      return result;
+      return result
     }
-    synchronized (BROWSER_COMPATIBLE_DATE_FORMAT_STRINGS) {
-      for (int i = 0, count = BROWSER_COMPATIBLE_DATE_FORMAT_STRINGS.length; i < count; i++) {
-        DateFormat format = BROWSER_COMPATIBLE_DATE_FORMATS[i];
-        if (format == null) {
-          format = new SimpleDateFormat(BROWSER_COMPATIBLE_DATE_FORMAT_STRINGS[i], Locale.US);
-          // Set the timezone to use when interpreting formats that don't have a timezone. GMT is
-          // specified by RFC 7231.
-          format.setTimeZone(UTC);
-          BROWSER_COMPATIBLE_DATE_FORMATS[i] = format;
-        }
-        position.setIndex(0);
-        result = format.parse(value, position);
-        if (position.getIndex() != 0) {
+    synchronized(BROWSER_COMPATIBLE_DATE_FORMAT_STRINGS) {
+      for (i in 0 until BROWSER_COMPATIBLE_DATE_FORMAT_STRINGS.size) {
+        val format: DateFormat = BROWSER_COMPATIBLE_DATE_FORMATS[i]
+            ?: SimpleDateFormat(BROWSER_COMPATIBLE_DATE_FORMAT_STRINGS[i], Locale.US).apply {
+              // Set the timezone to use when interpreting formats that don't have a timezone. GMT is
+              // specified by RFC 7231.
+              timeZone = UTC
+            }
+        BROWSER_COMPATIBLE_DATE_FORMATS[i] = format
```

Comment:
I don’t like how this adds an assignment when the value hasn’t changed. Could you make it an if instead?

Reply:
Fixed in 2664c76ddba226c8a6fbf510bc9af4ee2606eada